api/http-api.md

```diff
@@ -0,0 +1,173 @@
+      between describing an update and polling for an update is a "wait" boolean.
+* Activities:
+  * `RecordActivityTaskHeartbeat` - `POST /api/v1/namespaces/{namespace}/activities/heartbeat`
+    * We consider "heartbeat" a verb for these purposes
+  * `RecordActivityTaskHeartbeatById` - `POST /api/v1/namespaces/{namespace}/activities/heartbeat-by-id`
+    * We are intentionally not doing `/activities/{activity_id}/heartbeat` or similar due to URL ambiguity and that we
+      want to encourage the task token approach
+  * `RespondActivityTaskCompleted` - `POST /api/v1/namespaces/{namespace}/activities/complete`
+  * `RespondActivityTaskCompletedById` - `POST /api/v1/namespaces/{namespace}/activities/complete-by-id`
+  * `RespondActivityTaskFailed` - `POST /api/v1/namespaces/{namespace}/activities/fail`
+  * `RespondActivityTaskFailedById` - `POST /api/v1/namespaces/{namespace}/activities/fail-by-id`
+  * `RespondActivityTaskCanceled` - `POST /api/v1/namespaces/{namespace}/activities/cancel`
+  * `RespondActivityTaskCanceledById` - `POST /api/v1/namespaces/{namespace}/activities/cancel-by-id`
+* Search attributes:
+  * Intentionally not exposed high-level search attribute calls from workflow service
+* Task polling/completing:
+  * Intentionally not exposed (except for activity task stuff needed for async activities)
+* Task queues:
+  * `ResetStickyTaskQueue` - Intentionally not exposed
+  * `DescribeTaskQueue` - `GET /api/v1/namespaces/{namespace}/task-queues/{task_queue_name}`
+  * `ListTaskQueuePartitions` - Intentionally not exposed
+* Schedules:
+  * `CreateSchedule` - `POST /api/v1/namespaces/{namespace}/schedules/{schedule_id}`
+  * `DescribeSchedule` - `GET /api/v1/namespaces/{namespace}/schedules/{schedule_id}`
+  * `UpdateSchedule` - `POST /api/v1/namespaces/{namespace}/schedules/{schedule_id}/update`
+    * This is not a `PUT` because update different from create and not `PATCH` to avoid confusion with below
+  * `PatchSchedule` - `POST /api/v1/namespaces/{namespace}/schedules/{schedule_id}/patch`
+    * This is not a `PATCH` to avoid confusion with update above
+  * `ListScheduleMatchingTimes` - `GET /api/v1/namespaces/{namespace}/schedules/{schedule_id}/matching-times`
+  * `DeleteSchedule` - `DELETE /api/v1/namespaces/{namespace}/schedules/{schedule_id}`
+    * Confirm whether eventually consistent
+  * `ListSchedules` - `GET /api/v1/namespaces/{namespace}/schedules`
+* Batch operations:
+  * `StartBatchOperation` - `POST /api/v1/namespaces/{namespace}/batch-operations/{job_id}`
+  * `DescribeBatchOperation` - `GET /api/v1/namespaces/{namespace}/batch-operations/{job_id}`
+  * `ListBatchOperations` - `GET /api/v1/namespaces/{namespace}/batch-operations`
+  * `StopBatchOperation` - `POST /api/v1/namespaces/{namespace}/batch-operations/{job_id}/stop`
+* Worker build IDs:
+  * Intentionally not exposed until they settle
+* Other calls:
+  * `GetClusterInfo` - Intentionally not exposed
+  * `GetSystemInfo` - Intentionally not exposed
```

Comment:
We won't be able to use this API if this is not exposed.

Reply:
:+1: Will definitely expose then.

Reply:
Exposed